helm/revoke.go

```diff
@@ -85,54 +114,101 @@ func revokeAccessFromRBACEntities(
 ) error {
 	logger := logging.GetProjectLogger()
 
+	errList := NewRevocationErrors()
 	numEntities := len(rbacEntities)
 	for idx, rbacEntity := range rbacEntities {
 		logger.Infof("Revoking access from entity %s (%d of %d entities)", rbacEntity, idx+1, numEntities)
+		roleName := getTillerAccessRoleName(rbacEntity.EntityID(), tillerNamespace)
+
+		logger.Infof("Deleting entity role %s", roleName)
+		if err := deleteEntityRole(kubectlOptions, tillerNamespace, roleName, rbacEntity); err != nil {
+			errList.AddError(err)
+			logger.Warningf("Unable to delete role for entity %s from namespace %s", rbacEntity.Subject().Name, tillerNamespace)
+		} else {
+			logger.Infof("Deleted role for entity %s", rbacEntity.Subject().Name)
+		}
 
-		logger.Infof("Deleting entity role and binding")
-		if err := deleteEntityRoleAndBinding(kubectlOptions, tillerNamespace, rbacEntity); err != nil {
-			logger.Errorf("Unable to delete role for entity %s from namespace %s", rbacEntity.Subject().Name, tillerNamespace)
-			return err
+		logger.Infof("Deleting entity role binding")
+		if err := deleteEntityRoleBinding(kubectlOptions, tillerNamespace, roleName, rbacEntity); err != nil {
+			errList.AddError(err)
+			logger.Warningf("Unable to delete role binding for entity %s from namespace %s", rbacEntity.Subject().Name, tillerNamespace)
+		} else {
+			logger.Infof("Deleted role binding for entity %s", rbacEntity.Subject().Name)
 		}
 
 		logger.Infof("Deleting entity keypair from secrets")
 		if err := deleteEntityTLS(kubectlOptions, tillerNamespace, rbacEntity); err != nil {
+			errList.AddError(err)
 			logger.Errorf("Unable to delete TLS keypair for entity %s from namespace %s", rbacEntity.Subject().Name, tillerNamespace)
-			return err
 		}
+		logger.Infof("Deleted role binding for entity %s", rbacEntity.Subject().Name)
+	}
+	if !errList.IsEmpty() {
+		return errors.WithStackTrace(errList)
 	}
 	return nil
 }
 
-// deleteEntityRoleandBinding deletes the RBAC role and role binding associated with a provided entity
-func deleteEntityRoleAndBinding(
+// deleteEntityRole deletes the RBAC role associated with a provided entity
+func deleteEntityRole(
 	kubectlOptions *kubectl.KubectlOptions,
 	tillerNamespace string,
+	roleName string,
 	rbacEntity RBACEntity,
 ) error {
 	logger := logging.GetProjectLogger()
 
-	client, err := kubectl.GetKubernetesClientFromOptions(kubectlOptions)
+	labels := getTillerRoleLabels(rbacEntity.EntityID(), tillerNamespace)
+	filters := kubectl.LabelsToListOptions(labels)
+	roles, err := kubectl.ListRoles(kubectlOptions, tillerNamespace, filters)
+	if len(roles) == 0 {
+		return fmt.Errorf("Role not found: %s", err)
```

Comment:
NIT: Can you create a new error type here? Makes type checking the errors easier.